File: benchmarks/python/particles.py
N = 100000

particles = [
    {"x": float(i), "y": i * 2.0, "vx": 0.5, "vy": -0.5}
    for i in range(N)
]

for _ in range(100):
    for p in particles:
        p["x"] += p["vx"]
        p["y"] += p["vy"]

print(particles[49999]["x"])

File: benchmarks/ruby/particles.rb
N = 100_000

particles = Array.new(N) do |i|
  { x: i.to_f, y: i * 2.0, vx: 0.5, vy: -0.5 }
end

100.times do
  particles.each do |p|
    p[:x] += p[:vx]
    p[:y] += p[:vy]
  end
end

puts particles[49_999][:x]
